ethcore/src/lib.rs

```diff
@@ -150,6 +151,8 @@ pub mod snapshot;
 pub mod spec;
 pub mod state;
 pub mod state_db;
+// Test helpers made public for usage outside ethcore
+pub mod test_helpers;
```

Comment:
I tried that. Unfortunately there are a lot of uses of ethcore classes inside test_helpers. So it's not possible directly to move it to the separate crate and include as dependency for ethcore (because of cyclic dependencies). Implemented solution with public mod (I agree not the best one) was a compromise for complete refactoring of all the helpers

Reply:
than `ethcore` tests which is `test_helpers` should be moved to `ethcore/tests/` directory. And you can have a dependency cycle like this:

`test_helpers [dependencies] ` contains `ethcore`
`ethcore [dev-dependencies]` contains `test_helpers`
`private-transactions [dev-dependencies]` contains `test_helpers`

Reply:
Unfortunately, it seems, that Rust compiler doesn't allow such structure. The problem in details: let's say, we have struct A in ethcore crate and fn func() -> A in test_helpers crate. If to call func inside ethcore crate, the compiler returns the error: "expected struct A, found struct ethcore::A". Discussed the issue with @debris offline and decided to leave this as for now, other possible alternatives like massive copy-pasting of helpers seem worse.